### PR TITLE
enhance: Whether to enable mergeSort mode when performing mixCompaction

### PR DESCRIPTION
**Pull request by xiaocai2333**
issue: #37579

### CODE REVIEW

**Review by liliu-z — APPROVED** [review]
/lgtm

**Comment by liliu-z** [comment]
What's the relationship between this PR and the linked issue?

File: pkg/util/paramtable/component_param.go [hunk 1]
```diff
@@ -4215,6 +4215,7 @@ type dataNodeConfig struct {
 	// Compaction
 	L0BatchMemoryRatio       ParamItem `refreshable:"true"`
 	L0CompactionMaxBatchSize ParamItem `refreshable:"true"`
+	UseMergeSort             ParamItem `refreshable:"true"`
```
Comment (liliu-z) [inline pkg/util/paramtable/component_param.go:4218]: Let's make it false by default

Comment (xiaocai2333) [reply pkg/util/paramtable/component_param.go:4218]: ok

**Comment by xiaocai2333** [comment]
rerun ut

**Comment by xiaocai2333** [comment]
rerun cpp-unit-test

**Comment by xiaocai2333** [comment]
rerun go-sdk

**Comment by xiaocai2333** [comment]
rerun cpp-unit-test

**Comment by xiaocai2333** [comment]
rerun cpp-unit-test

**Comment by xiaocai2333** [comment]
rerun go-sdk

**Comment by xiaocai2333** [comment]
rerun ut

**Pull request merged**